Enterprise-CMCS/macpro-mdct-mfp
# Add cypress scaffolding

### Description

Cleanup cypress folder and re-initialize with basic setup and tests.

- Reorganize folder structure such that the test root sits at `tests/` instead of `tests/cypress` (to avoid duplicating the cypress sub-directory)
- Remove unused packages and update used ones
- Add yarn.lock file
- Add basic commands, config, and test files for cypress
  - Added login tests (these do not assert, just verify that they can reach the app and log in)
  - Added accessibility tests (these are skipped because they found issues, which should be resolved separately)
- Enabled the GitHub workflow to run E2E and A11y tests after branches deploy
  - Updated action versions and config as needed

### Related ticket(s)


CMDCT-3317

---
### How to test

Cypress:
- Pull this branch
- `cd tests/`
- `yarn`
- `CYPRESS_ADMIN_USER_PASSWORD=passwordhere CYPRESS_STATE_USER_PASSWORD=passwordhere yarn test`
- Validate it opens cypress
- Click "E2E" and then "Chrome"
- Run the login test and verify it passes

GitHub Workflow:
- Verify the tests run and pass in this PR!

(the accessibility tests are skipped. if you unskip them they uncover some accessibility issues for us to look into. this should be addressed separately)

---
### Author checklist


---
